specification/network/resource-manager/Microsoft.Network/stable/2018-02-01/expressRouteCrossConnection.json

```diff
@@ -0,0 +1,738 @@
+          "description": "The name of the primary  port."
+        },
+        "secondaryAzurePort": {
+          "type": "string",
+          "description": "The name of the secondary  port."
+        },
+        "sTag": {
+          "type": "integer",
+          "description": "The identifier of the circuit traffic."
+        },
+        "serviceProviderProperties": {
+          "$ref": "#/definitions/ExpressRouteCrossConnectionServiceProviderProperties",
+          "description": "The ServiceProviderProperties."
+        },
+        "expressRouteCircuit": {
+          "$ref": "#/definitions/ExpressRouteCircuitReference",
+          "description": "The ExpressRouteCircuit"
+        },
+        "serviceProviderProvisioningState": {
+          "type": "string",
+          "description": "The provisioning state of the circuit in the connectivity provider system. Possible values are 'NotProvisioned', 'Provisioning', 'Provisioned'.",
+          "enum": [
+            "NotProvisioned",
+            "Provisioning",
+            "Provisioned",
+            "Deprovisioning"
+          ],
+          "x-ms-enum": {
+            "name": "ServiceProviderProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "serviceProviderNotes": {
+          "type": "string",
+          "description": "Additional read only notes set by the connectivity provider."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Gets the provisioning state of the public IP resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        },
+        "peerings": {
+          "type": "array",
+          "items": {
+            "$ref": "./expressRouteCircuit.json#/definitions/ExpressRouteCircuitPeering"
+          },
+          "description": "The list of peerings."
+        }
+      },
+      "description": "Properties of ExpressRouteCrossConnection."
+    },
+    "ExpressRouteCrossConnection": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ExpressRouteCrossConnectionPropertiesFormat"
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+        "description": "ExpressRouteCrossConnection resource"
+    },
+    "ExpressRouteCrossConnectionListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ExpressRouteCrossConnection"
+          },
+          "description": "A list of ExpressRouteCrossConnection resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
```

Comment:
mark as readonly

Reply:
Should the nextLink record be marked readOnly ? A quick browse through some of the other files does not show nextLink marked as readOnly. Please advise and I will update.

Reply:
Yes, please mark this as readonly